### PR TITLE
Check the Front Office test with puppeteer - Smoke tests campaign

### PR DESCRIPTION
**Pull request by nesrineabdmouleh**
| Questions     | Answers
| ------------- | -------------------------------------------------------
| Branch?       | develop
| Description?  | Check the Front office test with puppeteer in Smoke tests campaign
| Type?         | refacto
| Category?     | TE
| BC breaks?    | no
| Deprecations? | no
| Fixed ticket? | no
| How to test?  | npm run specific-test smoke/crawlingFO

---
This change is [Reviewable](https://reviewable.io/reviews/prestashop/prestashop/14902)

### CODE REVIEW

File: tests/puppeteer/pages/FO/FO_cart.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
+const CommonPage = require('../commonPage');
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/FO/FO_cart.js:1]: why `FO/FO_` it's redundant.
Rename file please.

File: tests/puppeteer/pages/FO/FO_cart.js [hunk 2]
```diff
@@ -0,0 +1,35 @@
...
+
+module.exports = class FO_CART extends CommonPage {
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/FO/FO_cart.js:3]:
```suggestion
module.exports = class Cart extends CommonPage {
```

File: tests/puppeteer/pages/FO/FO_home.js [hunk 3]
```diff
@@ -0,0 +1,27 @@
+const CommonPage = require('../commonPage');
+
+module.exports = class FO_HOME extends CommonPage {
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/FO/FO_home.js:3]:
```suggestion
module.exports = class Home extends CommonPage {
```

File: tests/puppeteer/pages/FO/FO_product.js [hunk 4]
```diff
@@ -0,0 +1,34 @@
+const CommonPage = require('../commonPage');
+
+module.exports = class FO_PRODUCT extends CommonPage {
```
Comment (PierreRambaud) [inline tests/puppeteer/pages/FO/FO_product.js:3]:
```suggestion
module.exports = class Product extends CommonPage {
```

File: tests/puppeteer/test/data/FO/cart.js [hunk 5]
```diff
@@ -0,0 +1,9 @@
+
```
Comment (PierreRambaud) [inline tests/puppeteer/test/data/FO/cart.js]:
```suggestion
```

File: tests/puppeteer/pages/FO/FO_cart.js (hunk 1 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/FO/FO_cart.js:1]: These modifications will be done in a separate PR.

File: tests/puppeteer/pages/FO/FO_cart.js (hunk 2 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/FO/FO_cart.js:3]: These modifications will be done in a separate PR.

File: tests/puppeteer/pages/FO/FO_cart.js [hunk 6]
```diff
@@ -0,0 +1,35 @@
...
+  constructor(page) {
+    super(page);
+
+    // Selectors for cart page
+    this.productName = '#main li:nth-of-type(%NUMBER) div.product-line-info > a';
```
Comment (jolelievre) [inline tests/puppeteer/pages/FO/FO_cart.js:8]: In our recent javascript PRs (mainly on back office), we were asked to separate the selector and put them in a separate file Which seems like a very good idea to ease the reusability in E2E tests.
So I'm wondering if these selectors should be in a separate file as well?

File: tests/puppeteer/pages/FO/FO_home.js [hunk 7]
```diff
@@ -0,0 +1,27 @@
...
+   */
+  async checkHomePage() {
+    await this.page.waitForSelector(this.logoHomePage, {visible: true});
+  }
+
+  /**
+   * Go the product page
+   * @param id, product id
+   */
+  async goToProductPage(id) {
```
Comment (jolelievre) [inline tests/puppeteer/pages/FO/FO_home.js:23]: This seems like a method that will be needed in a lot of tests SHould this be in the `CommonPage` ?

File: tests/puppeteer/pages/commonPage.js [hunk 8]
```diff
@@ -57,4 +57,52 @@ module.exports = class CommonPage {
...
+  async checkTextValue(selector, textToCheckWith, parameter = 'equal') {
+    await this.page.waitForSelector(selector);
+    switch (parameter) {
+      case 'equal':
+        await this.page.$eval(selector, el => el.innerText)
+          .then(text => expect(text.replace(/\s+/g, ' ').trim()).to.equal(textToCheckWith));
+        break;
+      case 'contain':
+        await this.page.$eval(selector, el => el.innerText)
+          .then(text => expect(text).to.contain(textToCheckWith));
```
Comment (jolelievre) [inline tests/puppeteer/pages/commonPage.js:88]: Don't you need the same treatment as with `equal` and remove extra spaces? `text.replace(/\s+/g, ' ').trim()`

File: tests/puppeteer/pages/FO/FO_cart.js (hunk 6 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/FO/FO_cart.js:8]: That's a problem we've been struggling with. At the moment your implementation is incomplete and we need to go on. I think there is a way to get selectors from your files and inject them into our logic, but even if that was easily feasible we still will have the problem of the names. So when our implementation and your implementation are complete we'll have to rework everything anyway.

File: tests/puppeteer/pages/FO/FO_home.js (hunk 7 above)
Comment (SimonGrn) [reply tests/puppeteer/pages/FO/FO_home.js:23]: Like I understand it, it's a method to choose the nth product on the homepage. So it needs to be binded to the homepage.

File: tests/puppeteer/pages/commonPage.js (hunk 8 above)
Comment (nesrineabdmouleh) [reply tests/puppeteer/pages/commonPage.js:88]: No, the contain function check if the textToCheckWith contain text, then we don't need to remove extra spaces

**Review by matthieu-rolland — APPROVED** [review]
lgtm

**Pull request merged**